libs/blocks/icon-block/icon-block.js

```diff
@@ -14,44 +14,61 @@
 * Icon Block - v5.1
 */
 
-import { decorateBlockBg, decorateButtons } from '../../utils/decorate.js';
-import { createTag } from '../../utils/utils.js';
+import { decorateBlockText } from '../../utils/decorate.js';
 
-function decorateLayout(el) {
-  const children = el.querySelectorAll(':scope > div');
-  if (children.length > 1 && children[0].childNodes.length) {
-    decorateBlockBg(el, children[0]);
-  }
-  const foreground = createTag('div', { class: 'foreground' });
-  el.appendChild(foreground);
-  return foreground;
+const iconBlocks = {
+  small: {
+    fullwidth: ['M', 'M'],
+    vertical: ['S', 'M'],
+    centered: ['S', 'M'],
+    bio: ['S', 'S'],
+  },
+  medium: {
+    fullwidth: ['L', 'M'],
+    vertical: ['M', 'M'],
+    centered: ['M', 'M'],
+    bio: ['S', 'S'],
+  },
+  large: {
+    fullwidth: ['XL', 'M'],
+    vertical: ['M', 'M'],
+    centered: ['M', 'M'],
+    bio: ['S', 'S'],
+  },
+};
+
+function sortPriority(x, y) {
+  const priorities = ['bio', 'large', 'medium', 'small'];
+  let priority = 0;
+  if (priorities.includes(x)) priority = -1;
+  else if (priorities.includes(y)) priority = 1;
+  return priority;
+}
+
+function getBlockVariantSize(el) {
+  const attrs = [...el.classList];
+  const size = attrs.filter((i) => Object.keys(iconBlocks).includes(i)).sort(sortPriority)?.[0];
+  const sizeObj = iconBlocks[size];
+  const variant = attrs.filter((i) => Object.keys(sizeObj).includes(i)).sort(sortPriority)?.[0];
+  if (!size || !variant) return iconBlocks.large.fullwidth;
+  return iconBlocks[size][variant];
 }
 
-function decorateContent(row, isColumn) {
-  if (!row) return;
-  const text = row.querySelector('h1, h2, h3, h4, h5, h6')?.closest('div');
+function decorateContent(el) {
+  const block = el.querySelector(':scope > div:not([class])');
+  block.classList.add('foreground');
+  if (!block) return;
+  const text = block.querySelector('h1, h2, h3, h4, h5, h6, p')?.closest('div');
   if (text) {
-    text?.classList.add('text');
-    const headings = text?.querySelectorAll('h1, h2, h3, h4, h5, h6');
-    const heading = headings?.[headings.length - 1];
-    heading?.classList.add(isColumn ? 'heading-S' : 'heading-XL');
-    heading?.nextElementSibling?.classList.add('body-M');
-    heading?.previousElementSibling?.classList.add('icon-area');
-    decorateButtons(row);
-  } else {
-    row.classList.add('text');
-    const image = row.querySelector(':scope img');
+    text?.classList.add('text-content');
```

Comment:
Redundant since it's in the conditional
```suggestion
    text.classList.add('text-content');
```